### PR TITLE
Replace repr() with proper formatting

### PR DESCRIPTION
**Pull request by eumiro**
Refactor: replace `repr()` with `!r` in f-strings and `%r` in log messages.

### CODE REVIEW

File: airflow/jobs/backfill_job_runner.py [hunk 1]
```diff
@@ -1036,8 +1036,8 @@ def query(result, items):
 
         reset_tis = helpers.reduce_in_chunks(query, tis_to_reset, [], self.job.max_tis_per_query)
 
-        task_instance_str = "\n\t".join(repr(x) for x in reset_tis)
+        task_instance_str = "\n".join(f"\t{x!r}" for x in reset_tis)
         session.flush()
 
-        self.log.info("Reset the following %s TaskInstances:\n\t%s", len(reset_tis), task_instance_str)
+        self.log.info("Reset the following %s TaskInstances:\n%s", len(reset_tis), task_instance_str)
```
Comment (uranusjr) [inline airflow/jobs/backfill_job_runner.py:1042]: This seems unrelated?

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]
